functions/_fzf_search_git_status.fish

```diff
@@ -8,6 +8,7 @@ function _fzf_search_git_status --description "Search the output of git status.
             _fzf_wrapper --ansi \
                 --multi \
                 --query=(commandline --current-token) \
+                --preview='set -l path (string split " " {})[-1]; if not string match -r "\?\?.*" {}; if string match -r "^[NMAD]\s\s\S*" {}; git diff --color=always --cached -- $path; else if string match -r "^\s[NMAD]\s\S*" {}; git diff --color=always -- $path; else if string match -r "^[NMAD][NMAD]\s\S*" {}; git diff --color=always -- $path; git diff --color=always --cached -- $path; end; end' \
```

Comment:
Can you make this a helper function so it's easier to review and maintain? see the preview function for other features for examples